common/file_model/variant_allele.py

```diff
@@ -62,19 +62,27 @@ def traverse_csq_info(self) -> Mapping:
         info_map = {}
         for csq_record in self.variant.info["CSQ"]:
             csq_record_list = csq_record.split("|")
-            if csq_record_list[prediction_index_map["allele"]] not in info_map.keys():
-                info_map[csq_record_list[prediction_index_map["allele"]]] = {"phenotype_assertions": [], "predicted_molecular_consequences": [], "prediction_results": []} 
-
-            phenotypes = csq_record_list[prediction_index_map["phenotypes"]].split("&") if "phenotypes" in prediction_index_map.keys() else []   
-            for phenotype in phenotypes:
-                phenotype_assertions = self.create_allele_phenotype_assertion(phenotype) if phenotype else []
-                if (phenotype_assertions):
-                    info_map[csq_record_list[prediction_index_map["allele"]]]["phenotype_assertions"].append(phenotype_assertions)
+            allele = csq_record_list[prediction_index_map["allele"]]
+
+            if allele not in info_map.keys():
+                info_map[allele] = {"phenotype_assertions": [], "predicted_molecular_consequences": [], "prediction_results": []} 
+
+            # parse and form phenotypes - adding phenotype from any of the csq record would be enough for adding only variant-linked phenotypes
+            if not info_map[allele]["phenotype_assertions"]:
+                phenotypes = csq_record_list[prediction_index_map["phenotypes"]].split("&") if "phenotypes" in prediction_index_map.keys() else []   
```

Comment:
We are getting some duplicates for variant-linked phenotypes from data:
```suggestion
                phenotypes = csq_record_list[prediction_index_map["phenotypes"]].split("&") if "phenotypes" in prediction_index_map.keys() else []   
                phenotypes = list(set(phenotypes))  
```
This is still O(n) time complexity

Reply:
These should be resolved from the data files - to be done in next iteration of the data generation.